algebird-core/src/main/scala/com/twitter/algebird/Ring.scala

```diff
@@ -77,6 +85,14 @@ object ShortRing extends Ring[Short] {
   override def plus(l: Short, r: Short) = (l + r).toShort
   override def minus(l: Short, r: Short) = (l - r).toShort
   override def times(l: Short, r: Short) = (l * r).toShort
+  override def sum(t: TraversableOnce[Short]): Short = {
+    var sum: Short = 0
+    t.foreach { s => sum = (sum + s).toShort }
```

Comment:
is .toShort a method call or is it like a cast in java? 

maybe sum into an Int and then .toShort at the end?